### ISSUE TITLE
Plot training loss from log

### ISSUE THREAD

**Issue by sbrugman** [rbgirshick/py-faster-rcnn#347]
The caffe/tools/extras/parse_log.sh functionality is broken, as the logs of Faster R-CNN has a different format. It should be adapted to show plot the loss.

**Comment by rohitghosh** [comment]
Hi @sbrugman: You could use something like this where the log_path is the path of the log file

`def read_loss(self, log_path, mode):
        with open(log_path) as f:
            f = f.readlines()

        for i in range(1,5):
            if re.search('Iteration*',f[-i]) is not None:
                l = i
                break

        if mode == 'rpn-generate':
            bbox_line = f[-i-1]
            cls_line = f[-i-2]
            bbox_pattern = 'rpn_loss_bbox = [0-9.e]+ '
            cls_pattern = 'rpn_cls_loss = [0-9.e]+ '
        else:
            bbox_line = f[-i-2]
            cls_line = f[-i-1]
            bbox_pattern = 'bbox_loss = [0-9.e]+ '
            cls_pattern = 'cls_loss = [0-9.e]+ '

        try:
            output = re.search( bbox_pattern, bbox_line).group()
            bbox_loss = re.search ('[0-9.e]+', output).group()
        except:
            print ('Bbox loss couldn't be read from - {}'.format(bbox_line))
            #Pass a trash value so that while plotting a sharp drop in loss
            #can be interpreted as inablity to read the log
            bbox_loss = -100
        try:
            output = re.search( cls_pattern, cls_line).group()
            cls_loss = re.search ('[0-9.e]+', output).group()
        except:
            print ('Cls loss couldn't be read from - {}'.format(cls_line))
            cls_loss = -100`

The 'mode' argument was passed so as to differentiate between the different modes of loss output. You can work around by changing loss names and order from prototext file as well. You can pass this after every X iterations where X = display in the prototext file

**Comment by leon-woo** [comment]
I've modified the parse_log.py for faster_rcnn, but only for this project. Hers are my files, https://pan.baidu.com/s/1slaVUzV,  just copy these files to caffe-fast-rcnn/tools/extra/, and cover the original corresponding files.

**Comment by bailvwangzi** [comment]
@leon-woo The file `parse_log.sh` don't need to modify?

**Comment by tolry418** [comment]
Hi @rohitghosh : I am really wondering where can i get log file in faster rcnn folder? could you let me know more detail?!

**Comment by haoli-buaa** [comment]
@tolry418  Logs can be found in $FRCN_ROOT/experiments/logs/

**Comment by haoli-buaa** [comment]
I wrote a matlab script to generate train loss_curve
`clear;
clc;
close all;

% modify this according to your own log
train_log_file = 'logs/faster_rcnn_end2end_ZF_.txt.2017-11-15_22-07-29';
train_interval = 20;
test_interval = 20;

[~,string_output] = dos(['cat ', train_log_file, ' | grep ", loss = " | awk ''{print $9}''']);
train_loss = str2num(string_output);
n = 1:length(train_loss);
idx_train = (n-1) * train_interval;

figure;plot(idx_train, train_loss);
hold on;

grid on;
legend('Train Loss', 'Test Loss');
xlabel('iterations');
ylabel('loss');
title(' Train Loss Curve');`
